src/js/media-controller.js

```diff
@@ -532,17 +533,22 @@ const setAttr = async (child, attrName, attrValue) => {
     await delay(0);
   }
 
-  if (attrValue == undefined || (Array.isArray(attrValue) && !attrValue.length)) {
-    return child.removeAttribute(attrName);
+  // NOTE: For "nullish" (null/undefined), can use any setter
+  if (typeof attrValue === 'boolean' || attrValue == null) {
+    return setBooleanAttr(child, attrName, attrValue);
   }
-  if (typeof attrValue === 'boolean') {
-    if (attrValue) return child.setAttribute(attrName, '');
-    return child.removeAttribute(attrName);
+  if (typeof attrValue === 'number') {
+    return setNumericAttr(child, attrName, attrValue)
+  }
+  if (typeof attrValue === 'string') {
+    return setStringAttr(child, attrName, attrValue);
   }
-  if (Number.isNaN(attrValue)) {
+  // Treat empty arrays as "nothing" values
+  if (Array.isArray(attrValue) && !attrValue.length) {
     return child.removeAttribute(attrName);
   }
 
+  // For "special" values with custom serializers or all other values
   const val = CustomAttrSerializer[attrName]?.(attrValue) ?? attrValue;
```

Comment:
Should the check for a custom serializer not come before all the type checking?

Reply:
with the current impl, no, but your intuition here makes a lot of sense and maybe recommends some refactors in our projections. Gonna keep as is for now to not hold up the PR.